snap/container.go

```diff
@@ -196,20 +339,32 @@ func validateContainer(c Container, needsrx, needsx, needsr, needsf, noskipd map
 			return nil
 		}
 
-		if needsrx[path] || mode.IsDir() {
+		if mode&os.ModeSymlink != 0 && shouldValidateSymlink(path) {
+			symlinkInfo, err := evalAndValidateSymlink(c, path)
+			if err != nil {
+				logf("%s", err)
```

Comment:
I would add a context here, like `"Cannot evaluate and validate symbolic link: %s"`